### PR TITLE
[Merged by Bors] - chore: bump Std

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: lake-manifest.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
  [{"url": "https://github.com/leanprover/std4",
    "type": "git",
    "subDir": null,
-   "rev": "9e37a01f8590f81ace095b56710db694b5bf8ca0",
+   "rev": "baf6defee1fe881ae535519c0776f37f6ef08603",
```
Comment (eric-wieser) [inline lake-manifest.json:7]: https://github.com/leanprover/std4/compare/9e37a01f8590f81ace095b56710db694b5bf8ca0...baf6defee1fe881ae535519c0776f37f6ef08603

File: Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean [hunk 2]
```diff
@@ -89,7 +89,7 @@ lemma map_prod_map {α' : Type*} [MeasurableSpace α'] {β' : Type*} [Measurable
     (μ.map f).prod (ν.map g) = (μ.prod ν).map (Prod.map f g) := by
   apply Subtype.ext
   simp only [val_eq_toMeasure, toMeasure_prod, toMeasure_map]
-  rw [Measure.map_prod_map _ _ f_mble g_mble] <;> infer_instance
+  rw [Measure.map_prod_map _ _ f_mble g_mble]
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:92]: I can only assume this is somehow due to https://github.com/leanprover/std4/pull/407?

Comment (eric-wieser) [reply Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:92]: @digama0, does this make sense to you?

Comment (digama0) [reply Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:92]: this generates a linter warning on master

Comment (digama0) [reply Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:92]: I was curious to find out how this happened, and it seems that the LTAR file corresponding to this file was produced by the run https://github.com/leanprover-community/mathlib4/actions/runs/7105988780/job/19344435622 , which failed, but we commit the results anyway for files that built. Subsequent builds therefore skip this file, and so the noisy files lint does not fire.

Comment (eric-wieser) [reply Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:92]: Wow, that sounds like a pretty nasty problem with built-in linters +our CI setup

**Pull request closed without merging**